bat-utils/lib/runtime-kafka.js

```diff
@@ -21,29 +21,106 @@ class Kafka {
     this.runtime = runtime
     this.config = kafka
     this.topicHandlers = {}
+    this.topicConsumers = {}
   }
 
   async connect () {
-    // testing only
-    if (process.env.NODE_ENV === 'production') {
-      throw new Error('Kafka producer not allowed in production')
-    }
-
     const partitionCount = 1
 
-    this.producer = new NProducer(this.config, null, partitionCount)
-    this.producer.on('error', error => {
+    const producer = new NProducer(this.config, null, partitionCount)
+    this._producer = producer
+    producer.on('error', error => {
       console.error(error)
       if (this.runtime.captureException) {
         this.runtime.captureException(error)
       }
     })
-    await this.producer.connect()
+    await producer.connect()
+    // DEBUGGING
+    // const analytics = await producer.enableAnalytics()
+    // await new Promise((resolve) => {
+    //   producer.once("analytics", resolve)
+    // })
+    // const health = await producer.checkHealth()
+    // console.log(
+    //   analytics,
+    //   health
+    // )
+  }
+
+  async producer () {
+    if (this._producer) {
+      return this._producer
+    }
+    await this.connect()
```

Comment:
there should probably be some assertion that `this._producer` is set after this call. probably i'd format it like this

```
if (!this._producer) {
  await this.connect()
  assert(this._producer)
}
return this_.producer
```

Reply:
i don't think that this is something we have to worry about. creating a producer is happening sync
fixed in: eec22ca